File: (8 kyu) The Feast of Many Beasts/(8 kyu) The Feast of Many Beasts.js
// #1
// function feast(beast, dish) {
//   let result;
//   const start = beast[0] === dish[0];
//   const end = beast[beast.length - 1] === dish[dish.length - 1];
//   result = start && end;
//   return result;
// }
// #2
// function feast(beast, dish) {
//   let result;
//   const start = beast[0] === dish[0];
//   const end = beast.slice(-1) === dish.slice(-1);
//   result = start && end;
//   return result;
// }
// #3
const feast = (beast, dish) =>
  beast[0] === dish[0] && beast.slice(-1) === dish.slice(-1);

File: (8 kyu) The Feast of Many Beasts/(8 kyu) The Feast of Many Beasts.rs
fn feast(beast: &str, dish: &str) -> bool {
    // // #1
    //     let start = beast.starts_with(dish.chars().next().unwrap());
    //     let end = beast.ends_with(dish.chars().last().unwrap());
    //     return start && end;
    // // #2
    //     let start = beast.starts_with(&dish[0..1]);
    //     let end = beast.ends_with(&dish[dish.len()-1..]);
    //     start && end
    // #3
    &beast[..1] == &dish[..1] && &beast[beast.len() - 1..] == &dish[dish.len() - 1..]
}
